cmd/nodecmd/upgrade.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+package nodecmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/ava-labs/avalanche-cli/pkg/ansible"
+	"github.com/ava-labs/avalanche-cli/pkg/binutils"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanche-cli/pkg/ux"
+	"github.com/ava-labs/avalanche-cli/pkg/vm"
+	"github.com/spf13/cobra"
+	"golang.org/x/exp/slices"
+)
+
+func newUpgradeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "upgrade",
+		Short: "(ALPHA Warning) Update avalanchego or VM version for all node in a cluster",
+		Long: `(ALPHA Warning) This command is currently in experimental mode.
+
+The node update command suite provides a collection of commands for nodes to update
+their avalanchego or VM version.
+
+You can check the status after upgrade by calling avalanche node status`,
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+		RunE:         upgrade,
+	}
+
+	return cmd
+}
+
+func upgrade(_ *cobra.Command, args []string) error {
+	clusterName := args[0]
+	if err := setupAnsible(clusterName); err != nil {
+		return err
+	}
+	toUpgradeNodes, err := getNodesToBeUpgraded(clusterName)
+	if err != nil {
+		return err
+	}
+	for node, toUpgrade := range toUpgradeNodes {
+		// toUpgradeNodes either has value of avalancheGo or subnetEvm or avalancheGo,subnetEvm
+		toUpgradeItems := strings.Split(toUpgrade, ",")
+		for _, toUpGradeItem := range toUpgradeItems {
+			// toUpGradeItem either has value format of avalancheGo=<versionNum> or subnetEvm=<versionNum>
+			upgradeInfo := strings.Split(toUpGradeItem, "=")
```

Comment:
should we made a parallel update for all the cluster? of avago and subnet-evm. can we suppose
all the nodes in the cluster track the same vms?
if not, any way a parallel call can be made for all the nodes that have the same avago requirement, and also for all the nodes that have the same subnet-evm requirement. let's parallelize this